### PR TITLE
Add OneDeploy endpoint for AzureRmWebAppDeployment

### PR DESCRIPTION
**Pull request by dannysongg**
Add the logic to send deployment requests to the OneDeploy API for AzureRmWebAppDeployment to consume.

### CODE REVIEW

**Comment by dannysongg** [comment]
@vincentdass  @vsebesta  @rvairavelu

**Review by jvano — APPROVED** [review]

**Review by PhilipsonJoseph — APPROVED** [review]

**Comment by PhilipsonJoseph** [comment]
@dannysongg  we have Azure-rest-rm as a separate common package [https://github.com/microsoft/azure-pipelines-tasks-common-packages/blob/main/common-npm-packages/azure-arm-rest/azure-arm-app-service-kudu.ts]. Can you incorporate the changes here as well ?

File: common-npm-packages/azure-arm-rest/azure-arm-app-service-kudu.ts [hunk 1]
```diff
@@ -528,6 +528,41 @@ export class Kudu {
         }
     }
 
+    public async oneDeploy(webPackage: string, queryParameters?: Array<string>): Promise<any> {
+        let httpRequest = new webClient.WebRequest();
+        httpRequest.method = 'POST';
+        httpRequest.uri = this._client.getRequestUri(`/api/publish`, queryParameters);
```
Comment (PhilipsonJoseph) [inline common-npm-packages/azure-arm-rest/azure-arm-app-service-kudu.ts:534]: hi @dannysongg  I could see the checks are failing due to a build failure. i could see the wrong reference name used here. can you change **this_client** to **this.client**

**Review by PhilipsonJoseph — APPROVED** [review]

**Comment by jvano** [comment]
@manolerazvan

**Comment by v-mohithgc** [comment]
@dannysongg, please bump the package minor version to 235(current sprint) in both azure-arm-rest and azurermdeploycommon package.json file and Inside package folder build it with "npm i && npm run build".

Thank you.

**Pull request closed without merging**